sdks/python/apache_beam/io/gcp/datastore/v1new/helper.py

```diff
@@ -99,6 +105,19 @@ def write_mutations(batch, throttler, rpc_stats_callback, throttle_delay=1):
     rpc_stats_callback(throttled_secs=throttle_delay)
 
   try:
+    try:
+      batch.begin()
+    except ValueError:
+      # batch has already been started previously (and failed),
+      # we need to build it again from scratch (BEAM-7917)
+      batch = get_client(batch.project, batch.namespace).batch()
+      for mutation_type, mutation_element in mutations:
```

Comment:
*Maybe* we could reconstruct it from scratch by using [https://github.com/googleapis/google-cloud-python/blob/master/datastore/google/cloud/datastore/helpers.py#L105](https://github.com/googleapis/google-cloud-python/blob/master/datastore/google/cloud/datastore/helpers.py#L105) (or key_from_protobuf) assuming that we could get those from batch.mutations. Then we don't have to pass them explicitly but we will perform more encoding/decoding when errors occur.

Reply:
Most likely cannot, mutation_pb is different from entities/keys.